File: ds/sorting and hashing/extra/bubble.c
/*
Implement Bubble Sort
*/

#include<stdio.h>

void scanArray(int a[], int s)
{
    int i;
    printf("\n Enter %d numbers ", s);
    for(i =0 ; i< s; i++)
      scanf("%d", &a[i]);
}

void printArray(int a[], int s)
{
    int i;
    printf("\n");
    for(i =0 ; i< s; i++)
        printf(" %d ", a[i]);
}


void bubbleSort(int a[], int s)
{
    int i, j;//loop counter + array element index
    int temp; //swap
    
    for(i = s-1; i> 0; i--)
    {
        for(j =0 ; j< i; j++)
        {
            if(a[j] > a[j+1])// two adjacent elements are compared
            {//swap
                temp = a[j];
                a[j] = a[j+1];
                a[j+1] = temp;
            }//if
        }//for(j...
    }//for(i...
}

int main()
{
    int arr[6];
    
    scanArray(arr, 6);
    printArray(arr, 6);
    bubbleSort(arr,6);
    printArray(arr, 6);

    return 0;
}

File: ds/pythonstudy/bubble.py
#--------bubble sort------

def bubbleSort(arr, n):
 i = n-1
 while i > 0:
  for j in range(0,i):
   if arr[j] > arr[j+1]:
    temp = arr[j]
    arr[j] = arr[j+1]
    arr[j+1] = temp
  i = i-1

def scanArr(arr, n):
 for i in range(0,n):
  x = int(input("enter a number"))
  arr.append(x)
  
def sorting():
 arr=[]
 n = int(input("enter array size"))
 scanArr(arr,n)
 print(arr)
 bubbleSort(arr, n)
 print(arr)
